server/mergin/tests/test_project_controller.py

```diff
@@ -660,20 +654,15 @@ def test_update_project(client):
     assert project.get_role(test_user.id) is ProjectRole.READER
 
     # add tests user as writer to project
-    current_writers = [
-        u.id for u in project.project_users if u.role == ProjectRole.WRITER.value
-    ]
-    writers = [
-        u.username for u in User.query.filter(User.id.in_(current_writers)).all()
-    ]
-    data = {"access": {"writersnames": writers + [test_user.username]}}
+    data = {"access": {"writersnames": [test_user.username]}}
```

Comment:
I think in previous version was this access accepting whole array of names. If you defined one username, one username was saved to it, or?

Reply:
it still does accept whole array, but I had to modify the test because now if we have for example owner, there is no writer (and thus no writernames). Previously if you set one writer - you would end up with one writer (which is still the case now). 

The logic is following, whatever you specify in the list, will be set - either by upgrade/downgrade or adding new member. If you omit some key, that role will not by handled (only downgrade possible). If there is single id in multiple keys, the highest role wins.